app/controllers/application_controller/ci_processing.rb

```diff
@@ -672,6 +672,216 @@ def evacuate_form_fields
     }
   end
 
+  def associate_floating_ip_vms
+    assert_privileges("instance_associate_floating_ip")
+    recs = find_checked_items
+    recs = [params[:id].to_i] if recs.blank?
+    @record = find_by_id_filtered(VmOrTemplate, recs.first)
+    if @record.is_available?(:associate_floating_ip) && !@record.ext_management_system.nil?
+      if @explorer
+        associate_floating_ip
+        @refresh_partial = "vm_common/associate_floating_ip"
+      else
+        render :update do |page|
+          page << javascript_prologue
+          page.redirect_to :controller => 'vm',
+                           :action     => 'associate_floating_ip',
+                           :rec_id     => @record.id,
+                           :escape     => false
+        end
+      end
+    else
+      add_flash(_("Unable to associate %{floating_ip} with %{instance} \"%{name}\": %{details}") % {
+        :floating_ip => ui_lookup(:table => 'floating_ip'),
+        :instance    => ui_lookup(:table => 'vm_cloud'),
+        :name        => @record.name,
+        :details     => @record.is_available_now_error_message(:associate_floating_ip)}, :error)
+    end
+  end
+  alias instance_associate_floating_ip associate_floating_ip_vms
+
+  def associate_floating_ip
+    assert_privileges("instance_associate_floating_ip")
+    @record ||= VmOrTemplate.find_by_id(params[:rec_id])
```

Comment:
It might be better use Cloud Provider class instead of ```VmOrTemplate``` if Floating IP works only in Cloud (and ```:url``` few lines bellow references cloud too).